doc/source/architecture.rst

```diff
@@ -0,0 +1,213 @@
+additional running process associated with it. Here, only the CPython interface
+to the Mechanical API can be used.
+
+Distributed systems
+===================
+
+This section contains a very basic explanation of Distributed Systems. It is
+not meant be exhaustive and rigorous, but instead introduces just the topics
+necessary to understand the choices made by PyMechanical.
+
+The term Distributed System is used to describe a software system that uses
+a network to distribute software across multiple physical machines. With a
+Distributed System, the individual pieces of that system do not share an
+address space, and therefore cannot call functions of each other directly.
+Instead, they communicate with each other by sending messages in packets.
+
+There are unique characteristics of distributed systems relative to classical
+software systems that share an address space. For instance, in a distributed
+system, any call can fail because of a problem with the network, and the caller
+can not always know whether a call has failed. For this reason, features of
+interest to distributed system designers such as fault tolerance and
+idempotency are not necessarily important in classical software systems.
+
+These systems are used by email, multiplayer games, web applications, and high
+performance computing, among other things. Historically, there are three major
+categories of remote APIs, Message passing, resource-based, and Remote Method
+Invocation.
+
+Message passing
+---------------
+
+Message passing is the original model of remote APIs, and in fact the other two
+are implemented using message passing underneath. RPCs are initiated by the
+client and sent to the server as a message, or a sequence of bytes. There must
+be some understanding on both sides of the channel as to what information is in
+these bytes. This is usually handled by a protocol (such as HTTP or TCP) and
+usually also a higher level library (such as gRPC or zeroMQ) implemented using
+the protocol. The server may send back a series of messages to the client to
+indicate that a message was received, that it is executing, that it has
+executed, and with a response, if any. The protocol and library determine the
+expected sequence of messages, their ordering, and any failure handling.
+
+Resource-based APIs
+-------------------
+
+Resource-based APIs were popularized by REST, and has been shown to be, for
+practical purposes, infinitely scalable. They are traditionally implemented
+using HTTP or HTTPS but implementations can exist in any transport protocol.
+
+The fundamental concept behind REST is the separation of verbs and resources.
+Verbs include GET, PUT, UPDATE, DELETE, and POST, while resources are any
+uniquely identifiable entity.
+
+REST can scale because servers can make assumptions about the data it serves
+based on the verb and resource. For instance, if a GET was done on the resource
+"/a/b/c", and then no mutating verbs (PUT, UPDATE, DELETE, POST) have been run
```

Comment:
```suggestion
"/a/b/c", and then no mutating verbs (PUT, UPDATE, DELETE, POST) are run
```

Reply:
This isn't code, its a URI (uniform resource identifier). They are commonly expressed as strings. A subset or URIs are URLs, like the one pointing to this pull request

Reply:
We also tag URLs and file and directory names as code entities.